README.md

```diff
@@ -1,94 +1,187 @@
+||||
+|:point_right:| **Make Juju your own!** | Write your own charm: <br> https://juju.is/docs/sdk <br> _Juju is written in Go, but our SDK supports easy charm development in Python!_ |
 
-To learn more about Juju, visit https://juju.is/docs/olm.
+## Give it a try!
 
+Let's use `ops` to build a Kubernetes charm!
 
-## Pure Python
+### Set up
 
-The framework provides a standardised Python object model that represents the
-application graph, as well as an event-handling mechanism for distributed
-system coordination and communication.
+You will need Charmcraft, a Kubernetes cloud, and Juju. The quickest way is to use a Multipass VM launched with the `charm-dev` blueprint. 
 
-The latest version of ops requires Python 3.8 or above.
+Install Multipass: [Linux](https://multipass.run/docs/installing-on-linux) | [macOS](https://multipass.run/docs/installing-on-macos) | [Windows](https://multipass.run/docs/installing-on-windows). On Linux:
 
-Juju itself is written in Go for efficient concurrency even in large
-deployments. Charms can be written in any language, however, we recommend using
-Python with this framework to make development easier and more standardised.
-All new charms at Canonical are written using it.
+```
+sudo snap install multipass
+```
+
+Use Multipass to launch an Ubuntu VM with the `charm-dev` blueprint: 
+
+```
+multipass launch --cpus 4 --memory 8G --disk 30G --name tutorial-vm charm-dev 
+```
+
+Open a shell into the VM:
+
+```
+multipass shell tutorial-vm
+```
+
+Verify that you have Charmcraft, MicroK8s, and Juju:
+
+```
+snap list # should show all three
+```
+
+Verify that Juju recognizes MicroK8s as a localhost cloud:
+
+```
+juju clouds # should show both
+```
+
+Bootstrap a Juju controller into the MicroK8s cloud:
+
+```
+juju bootstrap microk8s tutorial-controller
+```
+
+Add a workspace, or 'model':
+
+```
+juju add-model tutorial-model
+```
+
+### Write your charm
 
+Create a charm directory and use Charmcraft to initialise your charm file structure:
 
-## Getting started
+```
+mkdir my-new-charm; cd my-new-charm; charmcraft init
+```
+This has created a standard charm directory structure. Poke around.
 
-A package of operator code is called a charmed operator or simply "charm".
-You'll use [charmcraft](https://juju.is/docs/sdk/install-charmcraft) to
-register your charm name and publish it when you are ready. You can follow one
-of our [charming tutorials](https://juju.is/docs/sdk/tutorials) to get started
-writing your first charm.
+Note that the `metadata.yaml` file shows that what we have is an example charm called `my-new-charm`, which uses an OCI image resource `httpbin` from `kennethreitz/httpbin`.
 
+Note that the `requirements.txt` file lists a version of `ops`. 
 
-## Testing your charms
+Note that the `src/charm.py` file imports `ops` and uses `ops` constructs to create a charm class `MyNewCharmCharm`, observe Juju events, and pair them to event handlers:
+
+```
+...
+import ops
+...
```

Comment:
```suggestion

import ops

```

Reply:
Ah, I missed that @benhoyt had made a similar suggestion to the one I just made (this comment was hidden in the files tab because it's on outdated changes). I also really don't like these.

I don't think we need to be explicit about the template file from `charmcraft init` having more content than what's being displayed. If you feel strongly about that then I would actually cut all of the top before the `class` line (`import ops` doesn't really add much to the tutorial and is implied by the use of `ops.X`) and if the end really needs to have something (I don't feel it does) then I think a `# Charm code continues` comment or similar would be better than the ellipsis (which has a different meaning in Python than in English).